x-pack/platform/plugins/shared/streams/server/routes/streams/significant_events/route.ts

```diff
@@ -201,15 +201,15 @@ const generateSignificantEventsRoute = createServerRoute({
         {
           inferenceClient,
           esClient: scopedClusterClient.asCurrentUser,
-          logger,
+          logger: logger.get('significant_events'),
           signal: getRequestAbortSignal(request),
         }
       )
     ).pipe(
-      mergeMap((queries) => fromRxjs(queries)),
-      map((query) => ({
-        query,
-        type: 'generated_query' as const,
```

Comment:
I wonder why it was implemented that way, it looks inefficient and should not have any benefit over the a single map. am i missing something ?